File: color_check.swift
import Foundation
import CoreGraphics
import CoreImage

guard CommandLine.arguments.count > 1 else { exit(1) }
let imagePath = CommandLine.arguments[1]
let url = URL(fileURLWithPath: imagePath)

guard let cgImage = CGImageSourceCreateWithURL(url as CFURL, nil).flatMap({ CGImageSourceCreateImageAtIndex($0, 0, nil) }) else {
    print("Could not load image")
    exit(1)
}

let width = cgImage.width
let height = cgImage.height
print("Image size: \(width)x\(height)")

guard let dataProvider = cgImage.dataProvider,
      let pixelData = dataProvider.data,
      let ptr = CFDataGetBytePtr(pixelData) else {
    print("Could not get pixel data")
    exit(1)
}

func getColor(u: Double, v_from_bottom: Double) {
    let px = Int(Double(width) * u)
    let py = Int(Double(height) * (1.0 - v_from_bottom))
    let bytesPerPixel = cgImage.bitsPerPixel / 8
    let bytesPerRow = cgImage.bytesPerRow
    let pixelInfo = py * bytesPerRow + px * bytesPerPixel
    
    let r = ptr[pixelInfo]
    let g = ptr[pixelInfo + 1]
    let b = ptr[pixelInfo + 2]
    
    print("Color at (\(u), \(v_from_bottom)) - px:\(px), py:\(py) -> R:\(r) G:\(g) B:\(b)")
}

getColor(u: 0.5, v_from_bottom: 0.57) // Create A Registry Background
getColor(u: 0.5, v_from_bottom: 0.47) // Find a Registry Background
getColor(u: 0.5, v_from_bottom: 0.35) // Between buttons and categories
getColor(u: 0.22, v_from_bottom: 0.30) // Cookware Background
getColor(u: 0.5, v_from_bottom: 0.88) // Registry Header Background

File: color_check.py
from PIL import Image
import sys

img_path = sys.argv[1]
try:
    img = Image.open(img_path)
    width, height = img.size
    print(f"Image size: {width}x{height}")
    
    # We know "Create A Registry" text is at x:0.19, y:0.57 (from bottom left, or wait... Vision is bottom-left origin)
    # Vision origin: bottom-left. y=0 is bottom, y=1 is top.
    # So y=0.57 in Vision means pixel y is height * (1 - 0.57) = height * 0.43 from top.
    
    def get_color(u, v_from_bottom):
        px = int(width * u)
        py = int(height * (1.0 - v_from_bottom))
        return img.getpixel((px, py))
    
    print("Background at Create A Registry (x:0.5, y:0.57):", get_color(0.5, 0.57))
    print("Background at Find A Registry (x:0.5, y:0.47):", get_color(0.5, 0.47))
    print("Background at Categories Grid (x:0.5, y:0.35):", get_color(0.5, 0.35))
    print("Background at Category 'Cookware' (x:0.22, y:0.30):", get_color(0.22, 0.30))
    print("Background at Registry Header (x:0.5, y:0.88):", get_color(0.5, 0.88))
except Exception as e:
    print(e)
